src/lib/massa-react/hooks/useWriteSmartContract.tsx

```diff
@@ -47,56 +51,19 @@ export function useWriteSmartContract(account: Provider, isMainnet = false) {
         fee,
       });
 
-      setOpId(operation.id);
-
-      loadingToastId = showToast(
-        'loading',
-        messages.pending,
-        operation.id,
-        isMainnet,
-      );
-
-      const op = new Operation(account, operation.id);
-      const finalStatus = await op.waitSpeculativeExecution();
-
-      toast.dismiss(loadingToastId);
-
-      setIsPending(false);
-      setIsOpPending(false);
-
-      if (finalStatus === OperationStatus.NotFound) {
-        setIsError(true);
-        showToast(
-          'success',
-          messages.timeout || Intl.t('steps.failed-timeout'),
-          operation.id,
-        );
-      } else if (
-        ![OperationStatus.SpeculativeSuccess, OperationStatus.Success].includes(
-          finalStatus,
-        )
-      ) {
-        logSmartContractEvents(account, operation.id);
-        throw new Error(`Operation failed with status: ${finalStatus}`);
-      } else {
-        setIsSuccess(true);
-        showToast('success', messages.success, operation.id);
-      }
-
-      return operation;
+      await executeOperation(operation, messages, final, isMainnet, setState);
```

Comment:
i dont really understand why this is called "executeOperation" maybe a better name can be found?!

Reply:
Ok I updated to processOperation let me know if you think it's better